src/units/abilities.hpp

```diff
@@ -39,7 +39,7 @@ struct individual_effect
 class effect
 {
 	public:
-		effect(const unit_ability_list& list, int def, const_attack_ptr attacker = const_attack_ptr(), bool is_cumulable = false);
+		effect(const unit_ability_list& list, int def, const_attack_ptr attacker = const_attack_ptr(), const std::string& type = "");
```

Comment:
How is this change connected to the aim of this PR?

Reply:
i prefer not if you don't know name for enum. In wait i let with ability name

Reply:
it's done

Reply:
it's done